client/src/views/TrackViewer/Layers.vue

```diff
@@ -101,47 +101,45 @@ export default defineComponent({
       const frame = frameNumber.value;
       const editingTrack = props.editingTrack.value;
       const selectedTrackId = props.selectedTrackId.value;
-      const trackIds = props.trackIds.value;
+      const tracks = props.tracks.value;
       const featurePointing = props.featurePointing.value;
       // Bug in interval search tree requires own return function for 0 values
-      const currentFrameIds = props.intervalTree.search(
+      const currentFrameIds: TrackId[] = props.intervalTree.search(
         [frame, frame],
         (value) => (value !== null ? value : null),
       );
       // Possibly include editing track or selected track
       // even if it's not in range.
       // if (sele!currentFrameIds.indexOf())
 
-      const tracks = [] as FrameDataTrack[];
+      const frameData = [] as FrameDataTrack[];
       const editingTracks = [] as FrameDataTrack[];
       currentFrameIds.forEach(
-        (item: number) => {
-          if (trackIds.includes(item)) {
-            if (props.trackMap.has(item)) {
-              const track = props.trackMap.get(item);
-              if (track === undefined) {
-                throw new Error(`trackMap missing trackid ${item}`);
-              }
-              const [features] = track.getFeature(frame);
-              const trackFrame = {
-                selected: (selectedTrackId === track.trackId),
-                editing: editingTrack,
-                trackId: track.trackId,
-                features,
-                confidencePairs: track.getType(),
-              };
-              tracks.push(trackFrame);
-              if (tracks[tracks.length - 1].selected && (editingTrack || featurePointing)) {
-                editingTracks.push(trackFrame);
-              }
+        (trackId: TrackId) => {
+          const track = props.trackMap.get(trackId);
+          if (track === undefined) {
+            throw new Error(`TrackID ${trackId} not found in map`);
+          }
+          if (tracks.includes(track)) {
+            const [features] = track.getFeature(frame);
+            const trackFrame = {
+              selected: (selectedTrackId === track.trackId),
+              editing: editingTrack,
+              trackId: track.trackId,
+              features,
+              confidencePairs: track.getType(),
+            };
+            frameData.push(trackFrame);
+            if (frameData[frameData.length - 1].selected && (editingTrack || featurePointing)) {
+              editingTracks.push(trackFrame);
```

Comment:
Think it would make sense in the future to just refactor this to pass directly to the drawing layers the filteredTracks and the `selectedTrackId/editingTrack props`?  Then when `editingTrack` is true we just do a `findIndexOf` instead of iterating over the entire filtered track list and pass that frameData to the editing layer.  It would avoid this data organizational loop of the filtered tracks at the expense of giving the layers more information and access to the Track items.